C++11 requires a space between literal and string macro

I'm building the current `master` branch with `g++ (GCC) 6.3.1 20170306`

```
mkdir build
cd build
cmake .. -DCMAKE_BUILD_TYPE=Debug
```

and I'm getting the following error:

```
[ 98%] Building CXX object drltrace/CMakeFiles/drltracelib.dir/drltrace.cpp.o
cd /home/vagrant/dev/drmemory/build/drltrace && /usr/bin/c++  -DASSEMBLE_WITH_GAS -DBUILD_NUMBER=1 -DDEBUG -DSTATISTICS -DTOOL_DR_MEMORY -DUNIX -DUSE_DRSYMS -DVERSION_COMMAS=1,11,17298 -DVERSION_NUMBER=1.11.17298 -DVERSION_STRING=\"1.11.17298\" -DX64 -Ddrltracelib_EXPORTS -I/home/vagrant/dev/drmemory/build/dynamorio/cmake -I/home/vagrant/dev/drmemory/build/dynamorio/cmake/../include -I/home/vagrant/dev/drmemory/build/dynamorio/ext/include -I/home/vagrant/dev/drmemory/common -I/home/vagrant/dev/drmemory/drmemory -I/home/vagrant/dev/drmemory/third_party/valgrind -I/home/vagrant/dev/drmemory/build/drmf/include -I/home/vagrant/dev/drmemory/drsyscall -I/home/vagrant/dev/drmemory/drsymcache -I/home/vagrant/dev/drmemory/umbra -I/home/vagrant/dev/drmemory/drfuzz  -fPIC    -DX86_64 -DLINUX -DDR_REG_ENUM_COMPATIBILITY -DDR_PAGE_SIZE_COMPATIBILITY -DDR_FAST_IR -DNOLINK_STRCASESTR -g -Wall -Werror -Wno-strict-aliasing -Wall -Werror -Wno-strict-aliasing -fno-stack-protector -o CMakeFiles/drltracelib.dir/drltrace.cpp.o -c /home/vagrant/dev/drmemory/drltrace/drltrace.cpp
In file included from /home/vagrant/dev/drmemory/drltrace/drltrace.cpp:48:0:
/home/vagrant/dev/drmemory/common/utils.h:469:22: error: invalid suffix on literal; C++11 requires a space between literal and string macro [-Werror=literal-suffix]
         NOTIFY_ERROR("ASSERT FAILURE (thread "TIDFMT"): %s:%d: %s (%s)" NL,  \
                      ^
/home/vagrant/dev/drmemory/common/utils.h:996:20: error: invalid suffix on literal; C++11 requires a space between literal and string macro [-Werror=literal-suffix]
 #define END_MARKER "\terror end"NL
                    ^
/home/vagrant/dev/drmemory/drltrace/drltrace.cpp:361:20: error: invalid suffix on literal; C++11 requires a space between literal and string macro [-Werror=literal-suffix]
         VNOTIFY(0, "drltrace log file is %s"NL, buf);
                    ^
```

--- a/common/utils.h
+++ b/common/utils.h
@@ -466,7 +466,7 @@ extern int tls_idx_util;
 #ifdef DEBUG
 # define ASSERT(x, msg) do { \
     if (!(x)) { \
-        NOTIFY_ERROR("ASSERT FAILURE (thread "TIDFMT"): %s:%d: %s (%s)" NL,  \
+        NOTIFY_ERROR("ASSERT FAILURE (thread " TIDFMT "): %s:%d: %s (%s)" NL,  \
                      (dr_get_current_drcontext() == NULL ? 0 :      \
                       dr_get_thread_id(dr_get_current_drcontext())),\
                      __FILE__,  __LINE__, #x, msg); \
@@ -993,7 +993,7 @@ drmem_strndup(const char *src, size_t max, heapstat_t type);
  * STRINGS
  */
 
-#define END_MARKER "\terror end"NL
+#define END_MARKER "\terror end" NL
 
 /* DR_MAX_OPTIONS_LENGTH is the maximum client options string length that DR
  * will give us.  by making each individual option buffer this long, we won't

--- a/drltrace/drltrace.cpp
+++ b/drltrace/drltrace.cpp
@@ -358,7 +358,7 @@ open_log_file(void)
                                           DR_FILE_ALLOW_LARGE,
                                           buf, BUFFER_SIZE_ELEMENTS(buf));
         ASSERT(outf != INVALID_FILE, "failed to open log file");
-        VNOTIFY(0, "drltrace log file is %s"NL, buf);
+        VNOTIFY(0, "drltrace log file is %s" NL, buf);
 
     }
 }

